src/routes/consent.ts

```diff
@@ -0,0 +1,271 @@
+import {
+  AcceptOAuth2ConsentRequestSession,
+  IdentityApi,
+  OAuth2ConsentRequest,
+} from "@ory/client"
+import { UserConsentCard } from "@ory/elements-markup"
+import bodyParser from "body-parser"
+import csrf from "csurf"
+import { defaultConfig, RouteCreator, RouteRegistrator } from "../pkg"
+import { register404Route } from "./404"
+import { oidcConformityMaybeFakeSession } from "./stub/oidc-cert"
+
+async function createOAuth2ConsentRequestSession(
+  grantScopes: string[],
+  consentRequest: OAuth2ConsentRequest,
+  identityApi: IdentityApi,
+): Promise<AcceptOAuth2ConsentRequestSession> {
+  // The session allows us to set session data for id and access tokens
+
+  const id_token: { [key: string]: any } = {}
+
+  if (consentRequest.subject && grantScopes.length > 0) {
+    const identity = (
+      await identityApi.getIdentity({ id: consentRequest.subject })
+    ).data
+
+    if (grantScopes.indexOf("email") > -1) {
+      // Client may check email of user
+      id_token.email = identity.traits["email"] || ""
+    }
+    if (grantScopes.indexOf("phone") > -1) {
+      // Client may check phone number of user
+      id_token.phone = identity.traits["phone"] || ""
+    }
+  }
+
+  return {
+    // This data will be available when introspecting the token. Try to avoid sensitive information here,
+    // unless you limit who can introspect tokens.
+    access_token: {
+      // foo: 'bar'
+    },
+
+    // This data will be available in the ID token.
+    id_token,
+  }
+}
+
+// A simple express handler that shows the Hydra consent screen.
+export const createConsentRoute: RouteCreator =
+  (createHelpers) => (req, res, next) => {
+    console.log("createConsentRoute")
+    res.locals.projectName = "An application requests access to your data!"
+
+    const { oauth2, identity } = createHelpers(req, res)
+    const { consent_challenge } = req.query
+
+    // The challenge is used to fetch information about the consent request from ORY hydraAdmin.
+    const challenge = String(consent_challenge)
+    if (!challenge) {
+      next(
+        new Error("Expected a consent challenge to be set but received none."),
+      )
+      return
+    }
+
+    let trustedClients: string[] = []
+    if (process.env.TRUSTED_CLIENT_IDS) {
+      trustedClients = String(process.env.TRUSTED_CLIENT_IDS).split(",")
+    }
+
+    console.log("getOAuth2ConsentRequest", challenge)
+    // This section processes consent requests and either shows the consent UI or
+    // accepts the consent request right away if the user has given consent to this
+    // app before
+    oauth2
+      .getOAuth2ConsentRequest({ consentChallenge: challenge })
+      // This will be called if the HTTP request was successful
+      .then(async ({ data: body }) => {
+        // If a user has granted this application the requested scope, hydra will tell us to not show the UI.
+        if (
+          body.skip ||
+          (body.client?.client_id &&
```

Comment:
What about `body.client?.skip_consent`? Now despite the client is created with `skip_consent: true,`, the consent screen would be shown.

Reply:
Yes, that's a possibility I didn't know about. I'll create a PR.